rampwf/workflows/simplified_image_classifier.py

```diff
@@ -0,0 +1,164 @@
+            have to contain image_classifier.py.
+        X_array : ArrayContainer vector of int
+            vector of image IDs to train on
+            (it is named X_array to be coherent with the current API,
+             but as said here, it does not represent the data itself,
+             only image IDs).
+        y_array : vector of int
+            vector of image labels corresponding to X_train
+        train_is : vector of int
+           indices from X_array to train on
+        """
+        folder, X_array = folder_X_array
+        if train_is is None:
+            train_is = slice(None, None, None)
+        submitted_image_classifier_file = '{}/{}.py'.format(
+            module_path, self.element_names[0])
+        image_classifier = imp.load_source('', submitted_image_classifier_file)
+        clf = image_classifier.ImageClassifier()
+        img_loader = ImageLoader(
+            X_array[train_is], y_array[train_is],
+            folder=folder,
+            n_classes=self.n_classes)
+        clf.fit(img_loader)
+        return clf
+
+    def test_submission(self, trained_model, folder_X_array):
+        """Test an image classifier.
+
+        trained_model : tuple (function, Classifier)
+            tuple of a trained model returned by `train_submission`.
+        X_array : ArrayContainer of int
+            vector of image IDs to test on.
+            (it is named X_array to be coherent with the current API,
+             but as said here, it does not represent the data itself,
+             only image IDs).
+        """
+        folder, X_array = folder_X_array
+        clf = trained_model
+        test_img_loader = ImageLoader(
+            X_array, None,
+            folder=folder,
+            n_classes=self.n_classes
+        )
+        y_proba = clf.predict_proba(test_img_loader)
+        return y_proba
+
+
+class ImageLoader(object):
+    """   
+    In image_classifier.py, both `fit` and `predict_proba` take as input
+    an instance of `ImageLoader`.
+    ImageLoader is used in `fit` and `predict_proba` to either load one image 
+    and its corresponding label  (at training time), or one image (at test time).
+    Images are loaded by using the method `load`.
+
+    Parameters
+    ==========
+
+    X_array : ArrayContainer of int
+        vector of image IDs to train on
+         (it is named X_array to be coherent with the current API,
+         but as said here, it does not represent the data itself,
+         only image IDs).
+
+    y_array : vector of int or None
+        vector of image labels corresponding to `X_array`.
+        At test time, it is `None`.
+
+    folder : str
+        folder where the images are
+
+    n_classes : int
+        Total number of classes. 
+    """
+
+    def __init__(self, X_array, y_array, folder, n_classes):
+        self.X_array = X_array
+        self.y_array = y_array
+        self.folder = folder
+        self.n_classes = n_classes
```

Comment:
this one is not used?

Reply:
Indeed, it is not used, but I thought that it coud be needed by users submissions to know the number of classes of the task, in order to to initialize the model.